lib/rex/ui/text/input/readline.rb

```diff
@@ -166,10 +166,10 @@ def readline_with_output(prompt, add_history=false)
           raise exception
         end
 
-        if add_history && line
+        if add_history && line && !line.to_s.start_with?(" ")
```

Comment:
Correct me if I'm wrong, but the `#to_s` here isn't necessary because if line is truthy, it will be a string. If it's not, it's going to fail on the nextline with `#strip` is called anyways.

```suggestion
        if add_history && line && !line.start_with?(' ')
```

Reply:
True this will be assigned to the result of `RbReadline.readline(reset_sequence + prompt)` as can be seen on line 161 of this file so this should be a text string or `nil` like you mentioned. Will apply this change so long.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15085/commits/db8cb32776918e6e52ccc8cd594074c9d829d078